helm/resource_release.go

```diff
@@ -840,15 +840,21 @@ func resourceDiff(ctx context.Context, d *schema.ResourceDiff, meta interface{})
 	if d.HasChanges(recomputeMetadataFields...) {
 		d.SetNewComputed("metadata")
 	}
-	if d.HasChange("version") {
-		// only recompute metadata if the version actually changes
-		// chart versioning is not consistent and some will add
-		// a `v` prefix to the chart version after installation
-		old, new := d.GetChange("version")
-		oldVersion := strings.TrimPrefix(old.(string), "v")
-		newVersion := strings.TrimPrefix(new.(string), "v")
-		if oldVersion != newVersion {
-			d.SetNewComputed("metadata")
+	_, err = os.Stat(d.Get("chart").(string))
+	_, err1 := os.Stat(d.Get("repository").(string))
+	_, err2 := url.ParseRequestURI(d.Get("chart").(string))
```

Comment:
Since this is repeated twice, you could put this logic in its own function. You could also then write a test for that function 😄 

Could maybe call the function`isLocalChart` ?